src/main/java/org/logstash/filters/geoip/GeoIPFilter.java

```diff
@@ -191,6 +191,11 @@ public boolean handleEvent(RubyEvent rubyEvent) {
       logger.debug("IP not found! exception={}, field={}, event={}", e, sourceField, event);
     } catch (GeoIp2Exception | IOException e) {
       logger.debug("GeoIP2 Exception. exception={}, field={}, event={}", e, sourceField, event);
+    } catch (NullPointerException e) {
```

Comment:
Thank's for suggesting I used an `IllegalArgumentException` littler fine grained than NPE, but it's not a specific exception for this use case. By now used only during retrieval of country, maybe we should cover with the same pattern all the other `retrieve*GeoData` because it could happen everywhere, WDYT?

Reply:
I prefer to close the gap in this PR, creating and using  a specialized exception instead of the IllegalArgument one. The exception would be part of this plugin, and used just inside it.

Reply:
This was fixed with latest commits